File: ArtikujtWindowsForms/ArtikujtWindowsForms/Models/Artikull.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace ArtikujtMVC.Models
{
    [Table("Artikujt", Schema = "dbo")]
    public class Artikull
    {
        [NotMapped]
        public bool IsNew { get; set; }
        public int Id { get; set; }
        [Required(ErrorMessage = nameof(Emri) +" duhet plotesuar")]
        public string Emri { get; set; }
        [Required(ErrorMessage = nameof(Njesia) +" duhet plotesuar")]
        public string Njesia { get; set; }
        [Required(ErrorMessage = "Data Skadences duhet plotesuar")]
        public DateTime DataSkadences { get; set; }
        [Range(0, int.MaxValue, ErrorMessage = "Cmimi nuk mund te jete nje numer negativ")]
        public double Cmimi { get; set; }
        [Required(ErrorMessage = nameof(Lloj) + " duhet plotesuar")]
        public string Lloj { get; set; }
        public bool KaTvsh { get; set; }
        [Required(ErrorMessage = nameof(Tipi) + " duhet plotesuar")]
        public string Tipi { get; set; }
        [Required(ErrorMessage = nameof(Barkod) + " duhet plotesuar")]
        public string Barkod { get; set; }
    }
}

File: ArtikullAndroid/app/src/main/java/com/example/artikujt/models/Artikull.java
package com.example.artikujt.models;

import java.util.Date;

public class Artikull {
    public boolean isNew;
    public int id;
    public String emri;
    public String njesia;
    public Date dataSkadences;
    public double cmimi;
    public Lloj lloj;
    public boolean kaTvsh;
    public Tipi tipi = Tipi.NONE;
    public String barkod;

    public Artikull(int id, String emri, String njesia, Date dataSkadences, double cmimi, Lloj lloj, boolean kaTvsh, Tipi tipi, String barkod) {
        this.id = id;
        this.emri = emri;
        this.njesia = njesia;
        this.dataSkadences = dataSkadences;
        this.cmimi = cmimi;
        this.lloj = lloj;
        this.kaTvsh = kaTvsh;
        this.tipi = tipi;
        this.barkod = barkod;
    }

    public Artikull(){}

    public void validate() throws InvalidStateException{
        if( isNullOrWhiteSpace(emri))
            throw  new InvalidStateException("Emri duhet plotesuar");
        if(isNullOrWhiteSpace(njesia))
            throw  new InvalidStateException("Njesia duhet plotesuar");
        if(dataSkadences == null)
            throw  new InvalidStateException("Data Skadences duhet plotesuar");
        if(lloj == null)
            throw  new InvalidStateException("Lloj duhet plotesuar");
        if(tipi == null || tipi == Tipi.NONE)
            throw  new InvalidStateException("Tipi duhet plotesuar");
        if(isNullOrWhiteSpace(barkod))
            throw  new InvalidStateException("Barkod duhet plotesuar");
    }

    private boolean isNullOrWhiteSpace(String str){
        return str == null || str.trim().isEmpty();
    }


    @Override
    public String toString() {
        return emri ;
    }
}
